packages/legend-graph/src/models/protocols/pure/v1/V1_PureGraphManager.ts

```diff
@@ -1734,6 +1733,9 @@ export class V1_PureGraphManager extends AbstractPureGraphManager {
           ),
       );
       const result = results[0];
+      if (result instanceof TestError) {
+        throw new IllegalStateError(result.error);
+      }
       assertType(result, TestFailed);
```

Comment:
Sorry could you explain why we need this assertion here?

Reply:
the way i generate expected result is run the test expecting to fail. 
Then use the diff to get expected result

Reply:
Did we document that somewhere, else could we add that here?